frameworks/php/phpunit/CodewarsListener.php

```diff
@@ -0,0 +1,54 @@
+<?php
+    use PHPUnit\Framework\TestCase;
+
+    class CodewarsListener implements PHPUnit_Framework_TestListener
+    {
+        public function addError(PHPUnit_Framework_Test $test, Exception $e, $time)
+        {
+            $message = preg_replace('/\n/g', '<:LF:>', $e->getMessage());
+            printf("<FAILED::>%s\n", $message);
+        }
+
+        public function addFailure(PHPUnit_Framework_Test $test, PHPUnit_Framework_AssertionFailedError $e, $time)
+        {
+            $message = preg_replace('/\n/g', '<:LF:>', $e->getMessage());
+            printf("<FAILED::>%s\n", $message);
+        }
+
+        public function addIncompleteTest(PHPUnit_Framework_Test $test, Exception $e, $time)
+        {
+            // printf("Test '%s' is incomplete.\n", $test->getName());
+        }
+
+        public function addRiskyTest(PHPUnit_Framework_Test $test, Exception $e, $time)
+        {
+            // printf("Test '%s' is deemed risky.\n", $test->getName());
+        }
+
+        public function addSkippedTest(PHPUnit_Framework_Test $test, Exception $e, $time)
+        {
+            // printf("Test '%s' has been skipped.\n", $test->getName());
+        }
+
+        public function startTest(PHPUnit_Framework_Test $test)
```

Comment:
I'm not familiar with how PHPUnit works but if the methods `startTest`, `endTest`, `startTestSuite` and `endTestSuite` are not used outside the class anyway, I think you can safely declare such methods as `protected` instead of `public`.